### PR TITLE
Added optimal travis ci settings

### PR DESCRIPTION
**Pull request by amilajack**
* Faster builds
* More stable tests (force npm installation to be latest version)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,13 +1,25 @@
...
   - 5
   - 6
 
+cache:
+  directories:
+    - node_modules
+
+install:
+  - npm i -g npm@latest
+  - npm install
```
Comment (medikoo) [inline .travis.yml]: What improvement that makes? Also global install doesn't make sense in case of this package as it doesn't expose any binaries

**Comment by medikoo** [comment]
Great thanks, if indeed that's an improvement I'll be happy to take it. I posted my doubts in comments

**Comment by amilajack** [comment]
> What improvement that makes?
`cache` delcaration tells travis to cache the node_modules directory. This makes builds much faster.

> Also global install doesn't make sense in case of this package as it doesn't expose any binaries
`npm I -g npm@latest` is the only global installation. What this line does is it overrides the default global version of npm and installs the latest version. Npm has a history of having bugs that made builds fail so its safest to use the one with the highest patch semvar.

**Comment by medikoo** [comment]
@amilajack thanks for clarifiction, I mistakenly assumed it's about global install of this package, which of course wasn't

**Comment by amilajack** [comment]
Thanks

**Pull request merged**